Content.Shared/_RMC14/OrbitalCannon/OrbitalCannonSystem.cs

```diff
@@ -165,7 +171,48 @@ private void OnWarheadPowerLoaderInteract(Entity<OrbitalCannonWarheadComponent>
 
     private void OnWarheadOrbitalBombardmentFire(Entity<OrbitalCannonWarheadComponent> ent, ref OrbitalBombardmentFireEvent args)
     {
-        Spawn(ent.Comp.Explosion, args.Coordinates);
+        var coordinates = _transform.ToCoordinates(args.Coordinates);
+
+        // chck for indestructible walls at impact location and try to find alternative
+        if (TileHasIndestructibleWalls(coordinates))
+        {
+            var found = false;
+            var adjacentOffsets = new Vector2i[]
+            {
+                new(-1, -1), new(0, -1), new(1, -1),  // Top row
+                new(-1,  0),             new(1,  0),  // Middle row (excluding center)
+                new(-1,  1), new(0,  1), new(1,  1)   // Bottom row
+            };
```

Comment:
You can make two nested for loops going from -1 to 1 for x and y so you don't need to create this array